src/utils/time.ts

```diff
@@ -1,10 +1,83 @@
+import { BATCH_TIME } from 'const'
+
 /**
  * Epoch in seconds
  */
 export function getEpoch(): number {
   return Math.floor(Date.now() / 1000)
 }
 
+/**
+ * Calculates the batchId. Either current batchId based on current Epoch
+ * or calculated if given a date
+ *
+ * Keep in mind this is used mainly for generating test data.
+ * The contract's `getBatchId` should be the source of truth.
+ *
+ * @param date? Optional Date object to calculate the batchId from.
+ *  Defaults to Date.now()
+ */
+export function calculateBatchId(date?: Date): number {
+  const timestamp = date ? date.getTime() : Date.now()
+  const timestampInSeconds = Math.floor(timestamp / 1000)
+  return Math.floor(timestampInSeconds / BATCH_TIME)
+}
+
+function adjustRollover(date: Date, newDate: Date): Date {
+  if (newDate.getDate() != date.getDate()) {
+    newDate.setDate(0)
+  }
+  return newDate
+}
+
+/**
+ * Adds time to a date. Modelled after MySQL DATE_ADD function.
+ * Example: dateAdd(new Date(), 30, 'minute')  //returns 30 minutes from now.
+ * https://stackoverflow.com/a/1214753/18511
+ *
+ * Adapted from https://stackoverflow.com/a/1214753/1272513
+ *
+ * @param date  Date to start with
+ * @param time  How much of given time unit to add to date.
+ * @param unit  One of: year, month, week, day, hour, minute, second
+ */
+export function addTimeToDate(
```

Comment:
Yes, this ideally uses a library. Here it looks there's some alternatives to momentjs if we feel moment might add to much weight.
https://github.com/you-dont-need/You-Dont-Need-Momentjs

Reply:
For this function I would consider OK not to use a lib, but I think we'll need to format dates easily, like for showing relative dates "fromNow" functionality

Reply:
Will indeed need that for the next step. Will look into it.